src/iblt.cpp

```diff
@@ -0,0 +1,293 @@
+/*
+Copyright (c) 2018 The Bitcoin Unlimited developers
+Copyright (c) 2014 Gavin Andresen
+
+Permission is hereby granted, free of charge, to any person obtaining a copy
+of this software and associated documentation files (the "Software"), to deal
+in the Software without restriction, including without limitation the rights
+to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+copies of the Software, and to permit persons to whom the Software is
+furnished to do so, subject to the following conditions:
+
+The above copyright notice and this permission notice shall be included in all
+copies or substantial portions of the Software.
+
+THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+SOFTWARE.
+*/
+#include "iblt.h"
+#include "hash.h"
+#include "iblt_params.h"
+#include <cassert>
+#include <iostream>
+#include <list>
+#include <sstream>
+#include <utility>
+
+static const size_t N_HASHCHECK = 11;
+
+template <typename T>
+std::vector<uint8_t> ToVec(T number)
+{
+    std::vector<uint8_t> v(sizeof(T));
+    for (size_t i = 0; i < sizeof(T); i++)
+    {
+        v.at(i) = (number >> i * 8) & 0xff;
+    }
+    return v;
+}
+
+
+bool HashTableEntry::isPure() const
+{
+    if (count == 1 || count == -1)
+    {
+        uint32_t check = MurmurHash3(N_HASHCHECK, ToVec(keySum));
+        return (keyCheck == check);
+    }
+    return false;
+}
+
+bool HashTableEntry::empty() const { return (count == 0 && keySum == 0 && keyCheck == 0); }
+void HashTableEntry::addValue(const std::vector<uint8_t> v)
+{
+    if (v.empty())
+    {
+        return;
+    }
+    if (valueSum.size() < v.size())
+    {
+        valueSum.resize(v.size());
+    }
+    for (size_t i = 0; i < v.size(); i++)
+    {
+        valueSum[i] ^= v[i];
+    }
+}
+
+CIblt::CIblt()
+{
+    valueSize = 0;
```

Comment:
What is valueSize used for?  Where is it used (I do not see a use)

Reply:
It looks like valueSize is used in an assert on line 118 to ensure that items inserted into the IBLT have the correct size as specified during object construction.